### DISCUSSION TITLE
Can Gatsby generate million pages?

### DISCUSSION THREAD

**Discussion by tombohub** [gatsbyjs/gatsby#28780] [Help]
I have a approximately about 1 million of json files, and foreign key related json files,  I need to turn into static pages.

Can I do that with Gatsby and how long do you think it would last?

What do you suggest how to do it?

**Comment by KyleAMathews** [comment]
We have a benchmark site for JSON you can check out — https://github.com/gatsbyjs/gatsby/tree/master/benchmarks/gabe-json-text

Depending on how your data is setup, it may or might not make sense to use gatsby-transformer-json like that site does. The other approach would be to load the data in your gatsby-node.js and add the data directly via `context` (this page talks in more detail about how to do this if you're unfamiliar — https://www.gatsbyjs.com/docs/how-to/querying-data/using-gatsby-without-graphql/)
